osgi.enroute.examples.rest.test/src/osgi/enroute/examples/rest/test/RestExamplesTest.java

```diff
@@ -0,0 +1,233 @@
+    	HttpClient client = new HttpClient();
+
+    	URI url = new URI("http://localhost:8080/rest/upper/steak");
+
+    	String result = client.build().get(String.class).go(url);
+
+    	Assert.assertEquals("\"STEAK\"", result);
+
+    	client.close();
+    }
+
+    @Test
+    public void testGetUpperWithTwoArgs() throws Exception {
+    	HttpClient client = new HttpClient();
+
+    	URI url = new URI("http://localhost:8080/rest/upper/steak1/steak2");
+
+    	String result = client.build().get(String.class).go(url);
+
+    	Assert.assertEquals("\"STEAK1 and STEAK2\"", result);
+
+    	client.close();
+    }
+
+    @Test
+    public void testGetUpper2() throws Exception {
+    	HttpClient client = new HttpClient();
+
+    	URI url = new URI("http://localhost:8080/rest/upper2/?upper2=test");
+
+    	String result = client.build().get(String.class).go(url);
+
+    	Assert.assertEquals("\"TEST\"", result);
+
+    	client.close();
+    }
+
+    public static class History extends DTO {
+		public String input;
+		public String output;
+	}
+
+    @Test
+    public void testGetUpper3() throws Exception {
+    	HttpClient client = new HttpClient();
+
+    	URI url = new URI("http://localhost:8080/rest/upper3/TesT");
+
+    	HttpRequest<History> httpRequest = client.build().get(History.class);
+
+		History result = httpRequest.go(url);
+
+    	Assert.assertNotNull(result);
+
+    	Assert.assertEquals("TesT", result.input);
+
+    	Assert.assertEquals("TEST", result.output);
+
+    	url = new URI("http://localhost:8080/rest/upper3/TesT2");
+
+		result = httpRequest.go(url);
+
+    	Assert.assertNotNull(result);
+
+    	Assert.assertEquals("TesT2", result.input);
+
+    	Assert.assertEquals("TEST2", result.output);
+
+    	client.close();
+    }
+
+    @Test
+    public void testDeleteUpper3() throws Exception {
+    	HttpClient client = new HttpClient();
+
+    	URI url = new URI("http://localhost:8080/rest/upper3/TesT2");
+
+    	Object result = client.build().delete().go(url);
+
+    	Assert.assertNotNull(result);
+
+    	client.close();
+    }
+
+    @Test
+    public void testPostUpper4() throws Exception {
+    	HttpClient client = new HttpClient();
+
+    	URI url = new URI("http://localhost:8080/rest/upper4/Sam");
+
+    	HttpRequest<Object> request = client.build().post();
+
+		Object result = request.go(url);
+
+    	Assert.assertEquals("{\"input\":\"Sam\",\"output\":\"SAM\"}", result);
+
+    	url = new URI("http://localhost:8080/rest/upper4/Sam2");
+
+    	result = request.go(url);
+
+    	Assert.assertEquals("{\"input\":\"Sam2\",\"output\":\"SAM2\"}", result);
+
+    	client.close();
+    }
+
+    @Test
+    public void testPutUpper4() throws Exception {
+    	HttpClient client = new HttpClient();
+
+    	URI url = new URI("http://localhost:8080/rest/upper4/Sam");
+
+    	HttpRequest<Object> request = client.build().put();
+
+		Object result = request.go(url);
+
+    	Assert.assertEquals("{\"input\":\"Sam\",\"output\":\"SAM\"}", result);
```

Comment:
The test is wrong. Instead of using the dtos encoder (or JSONCodec in bnd on which it is build) to compare it uses the toString() method which is not proper JSON. Both encoders have canonicalized outputs. You can also of course test the fields.